### ISSUE TITLE
weird startup scaling issues on macOS after winit `0.30`

### ISSUE THREAD

**Issue by tychedelia** [bevyengine/bevy#13702]
See video here:

https://github.com/bevyengine/bevy/assets/10366310/fa4dbd1d-03c6-4e40-924f-ba99858e3d76

This probably doesn't need to be blocking for `0.14` but feels pretty gross.

**Comment by rparrett** [comment]
It seems like the initial render is at `scale_factor` `1.0` and updates to the correct scale factor on the next frame.

**Comment by rparrett** [comment]
No surprise, but bisected to #13366

**Comment by alice-i-cecile** [comment]
Yeah, strongly suspect we're not setting the initial value correctly. If y'all can figure out a fix, ping me and I'll merge it for 0.14.

**Comment by rparrett** [comment]
I think that what is generally going on is:

- We ask winit to create a window. We have no way of knowing what the scale factor is if we didn't override it.
- winit creates it and sends a resize event with the correct size / scale factor
- we don't respond to that event until a frame later.

I am not sure if this needs to be split off into a separate issue yet (will try to test on another platform), but I am also seeing `scale_factor_override` not yielding the correct window size.

Adding
```
primary_window: Some(Window {
    resolution: WindowResolution::default().with_scale_factor_override(2.0),
    ..default()
}),
```
Results in fonts becoming larger, but the window stays the same size. (forever, it never gets resized)
